optuna/storages/rdb/storage.py

```diff
@@ -437,7 +437,14 @@ def set_trial_system_attr(self, trial_id, key, value):
         session = self.scoped_session()
 
         trial = models.TrialModel.find_or_raise_by_id(trial_id, session)
-        self.check_trial_is_updatable(trial_id, trial.state)
+        if key == '_number':
+            # `_number` attribute may be set even after a trial is finished.
+            # This happens if the trial was created before v0.9.0,
+            # where a trial didn't have `_number` attribute.
+            # In this case, `check_trial_is_updatable` is skipped to avoid the `RuntimeError`.
```

Comment:
I think this procedure will be not necessary when we add `number` attribute to `TrialModel`. If so, could you add a TODO comment to remove it?

Reply:
Thank you for your suggestion. I added a TODO comment (https://github.com/pfnet/optuna/pull/415/commits/2c72a8f6f946d079f825a5e303f6d585d84be114).